server/src/main/java/org/elasticsearch/action/admin/cluster/allocation/TransportGetAllocationStatsAction.java

```diff
@@ -42,13 +43,24 @@
 import java.io.IOException;
 import java.util.EnumSet;
 import java.util.Map;
+import java.util.concurrent.atomic.AtomicReference;
 
 public class TransportGetAllocationStatsAction extends TransportMasterNodeReadAction<
     TransportGetAllocationStatsAction.Request,
     TransportGetAllocationStatsAction.Response> {
 
     public static final ActionType<TransportGetAllocationStatsAction.Response> TYPE = new ActionType<>("cluster:monitor/allocation/stats");
 
+    public static final long CACHE_DISABLED = 0L;
+    public static final Setting<TimeValue> CACHE_MAX_AGE_SETTING = Setting.timeSetting(
+        "cluster.transport.get.allocation.stats.action.cache.max_age",
+        TimeValue.timeValueMinutes(1),
+        TimeValue.timeValueMillis(CACHE_DISABLED),
```

Comment:
Maybe just `TimeValue.ZERO` for the min, no need for a new constant here I think.